includes/classes/DistributorPost.php

```diff
@@ -314,7 +335,12 @@ public function get_the_id() {
 	 * @return string Post permalink.
 	 */
 	public function get_permalink() {
-		return get_permalink( $this->post );
+		static $permalink = null;
```

Comment:
Not opposed to using statics here but does this actually help in practice? Are we calling these methods multiple times in the same execution?

Reply:
We may... some of the canonical filters call `get_permalink`.

I'm not sure what's better or worse: making sure the site matches and switching if needs be or using statics. The former is slow but the latter may hit order of operations issues if other plugins add or remove filters.